test/functional/test_framework/test_node.py

```diff
@@ -187,7 +187,11 @@ def wait_for_rpc_connection(self):
                 if e.errno != errno.ECONNREFUSED:  # Port not yet open?
                     raise  # unknown IO error
             except JSONRPCException as e:  # Initialization phase
-                if e.error['code'] != -28:  # RPC in warmup?
+                if e.error['code'] == -28:  # RPC in warmup?
+                    pass
+                elif e.error['code'] == -342: # RPC handler unregistered, but http server haven't been stopped yet.
+                    raise FailedToStartError(self._node_msg(e.error['message']))
```

Comment:
Would it be good to call self.process.wait() here and include the exit code in the exception, so this is more similar to the previous FailedToStartError exception above?

Reply:
I would prefer @MarcoFalke's method, then we don't have to raise it here again.